src/cmdstan/stansummary_helper.hpp

```diff
@@ -332,6 +329,12 @@ stan::mcmc::chains<> parse_csv_files(const std::vector<std::string> &filenames,
   std::ifstream ifstream;
   ifstream.open(filenames[0].c_str());
   stan::io::stan_csv stan_csv = stan::io::stan_csv_reader::parse(ifstream, out);
+  if (stan_csv.samples.rows() < 1) {
```

Comment:
I deleted all the samples in a file and got a segfault:

```
bbales2@tadpole:~/cmdstan-develop$ ./bin/stansummary output2.csv
Warning: non-fatal error reading samples
terminate called after throwing an instance of 'std::invalid_argument'
  what():  No sampling draws found in Stan CSV file: output2.csv.
Aborted (core dumped)
```

Reply:
Similarly if I deleted the csv header I get a segfault:

```
bbales2@tadpole:~/cmdstan-develop$ ./bin/stansummary output3.csv
Warning: non-fatal error reading metadata
Error: error reading header
terminate called after throwing an instance of 'std::invalid_argument'
  what():  Error with header of input file in parse
Aborted (core dumped)
```

Seems like the errors are getting detected but there's still something weird happening with memory.

Reply:
good catch.  looking into it now.

Reply:
OK, there was an uncaught exception.  tightened up file open/close, added more checks.